src/cluster_legacy.h

```diff
@@ -91,6 +93,10 @@ typedef struct clusterNodeFailReport {
 #define CLUSTERMSG_TYPE_MODULE 9                /* Module cluster API message. */
 #define CLUSTERMSG_TYPE_PUBLISHSHARD 10         /* Pub/Sub Publish shard propagation */
 #define CLUSTERMSG_TYPE_COUNT 11                /* Total number of message types. */
+#define CLUSTERMSG_MSB 0x8000
```

Comment:
`MSB` conveys just the position but not the more important semantic information
```suggestion
#define CLUSTERMSG_LIGHT 0x8000
```

Reply:
Where are the light pubsub and light shard pubsub messages defined? Are they not used anymore?

I didn't think think change should imply any refactoring other than using the numbers 0x8000 and 0x8001 for light publish and light puvlish shard, but now it seems the logic was rewritten in some other way.

Is light pubsub now CLUSTERMSG_TYPE_PUBLISH | CLUSTERMSG_LIGHT?

Keep in mind that not all messages can be light. Only some. So using implicit logic like that can accidentally allow things like liggt ping?

Reply:
```suggestion
#define CLUSTERMSG_LIGHT_PUBLISH 0x8000
#define CLUSTERMSG_LIGHT_PUBLISHSHARD 0x8001
```

Ignore me if I missed some discussion about this.

Reply:
The problem with this is that it will get complicated to maintain the stats. We check with the number of commands and also the validity of the type is checked if the type is less that the number or types. Its all over the code and will lead to more changes.

> Is light pubsub now CLUSTERMSG_TYPE_PUBLISH | CLUSTERMSG_LIGHT?
Yes, we would be using less types and set the MSB for the light header.

What do you think?
Too extreme?

Reply:
>I didn't think think change should imply any refactoring other than using the numbers 0x8000 and 0x8001 for light publish and light puvlish shard, but now it seems the logic was rewritten in some other way.

From the last discussion above, I thought we're getting rid of the multi-message support. I can add it back.

Reply:
`CLUSTERMSG_LIGHT` can be considered as a property and not as a different type, we can add safety nets as to prevent other msgs using it.

Reply:
I don't know if it's "extreme". :)

If only the header would differ and the payload would be identical for publish and light-publish, then this would be logical, but there are other differences in the payload too.

Furthermore, don't we need some way to mark which messages are allowed as light and as heavy? We need to protect ourselves from crashing if a malicious node sends "light pong" and other invalid light-flagged stuff, right? Or did you add this already? (I didn't check)

Reply:
Did we get rid of multi-message completely now? Sorry, I didn't follow. The last time I checked, we didn't make a decision. We just discussed and had different opinions about it.

Reply:
The only decision I've seen is for the issue #557.

Reply:
> Did we get rid of multi-message completely now? Sorry, I didn't follow. The last time I checked, we didn't make a decision. We just discussed and had different opinions about it.

Not completely removed. We had a discussion in today's core meeting. The decision was to future proof the format but not adding new logic to handle true multi-pub messages. Do you have any concerns?

Reply:
> We need to protect ourselves from crashing if a malicious node sends "light pong" and other invalid light-flagged stuff, right? Or did you add this already? (I didn't check)

I did add an assert, Though I am guessing we can ignore these messages and logging it rather than crashing.

Reply:
Sorry I missed the meeting today.



> not adding new logic to handle true multi-pub messages. Do you have any concerns?

Yes I do, as I explained in another comment:

> if we can't receive messages with count > 1, then a future valkey can't send such messages, without adding a bunch of more logic to check whick nodes can receive light with only count == 1, count >= 1 or only heavy, so still the same amount of logic.

If we do this, then we can as well remove the count field completely. I don't like this, but if this is what you decided...

Reply:
Update: No concerns! I'm convinced after seeing Madelyn's comment

> End users should not start using the mpublish style commands until all nodes in the cluster are updated.